ledger-core/virtual/integration/staterequest_test.go

```diff
@@ -110,9 +111,8 @@ func TestVirtual_VStateRequest(t *testing.T) {
 			server.SendMessage(ctx, msg)
 
 			commontestutils.WaitSignalsTimed(t, 10*time.Second, waitVStateReport)
-			commontestutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
 
-			require.Equal(t, 1, typedChecker.VStateReport.Count())
+			require.Equal(t, 1, typedChecker.VStateReport.CountBefore())
```

Comment:
I wonder why you have to use CountBefore? Handler could leak after? Why WaitAllAsyncCallsDone is removed?

Reply:
updated